### PR TITLE
Feature/config extension

### PR DESCRIPTION
**Pull request by jmezna**

### CODE REVIEW

File: common/src/main/java/com/kumuluz/ee/configuration/utils/ConfigurationUtil.java [hunk 1]
```diff
@@ -153,11 +156,19 @@ public void set(String key, Float value) {
...
+
+        watch(key);
+
     }
 
     public void unsubscribe(ConfigurationListener listener) {
         config.getDispatcher().unsubscribe(listener);
     }
+
+    public void notifyChange(String key, String value) {
```
Comment (TFaga) [inline common/src/main/java/com/kumuluz/ee/configuration/utils/ConfigurationUtil.java]: Remove this method in the util in order to not allow anyone to publish changes to the config. Instead pass the ConfigurationDispatcher instance to the ConfigurationSource and make sure the source calls notify on that instance.

File: core/src/main/java/com/kumuluz/ee/EeApplication.java [hunk 2]
```diff
@@ -106,13 +110,27 @@ private void initialize() {
...
-        // Initialize ConfigurationDispatcher
-        // getPropery()
-        // Insert into ConfigurationUtil
+        log.info("Initializing config extensions");
+
+        List<Extension> configExtensions = extensions.stream().filter(extension -> extension.getClass()
+                .getDeclaredAnnotation(EeExtensionDef.class).type().equals(EeExtensionType.CONFIG)).collect
+                (Collectors.toList());
+
+        for (Extension configExtension : configExtensions) {
```
Comment (TFaga) [inline core/src/main/java/com/kumuluz/ee/EeApplication.java]: Additionally check if the extension contains a list of required Ee dependencies and validate if they are fulfilled.

File: core/src/main/java/com/kumuluz/ee/EeApplication.java [hunk 3]
```diff
@@ -163,6 +181,16 @@ private void initialize() {
         log.info("Components initialized");
 
         // Initiate the other extensions (filter(c.type -> c != CONFIG))
+        List<Extension> otherExtensions = extensions.stream().filter(extension -> !extension.getClass()
+                .getDeclaredAnnotation(EeExtensionDef.class).type().equals(EeExtensionType.CONFIG)).collect
+                (Collectors.toList());
+
+        log.info("Initialising non-config extensions");
+        for (Extension extension : otherExtensions) {
```
Comment (TFaga) [inline core/src/main/java/com/kumuluz/ee/EeApplication.java]: Same as above for the config extensions.

File: common/src/main/java/com/kumuluz/ee/configuration/utils/ConfigurationUtil.java (hunk 1 above)
Comment (jmezna) [inline common/src/main/java/com/kumuluz/ee/configuration/utils/ConfigurationUtil.java]: fixed

File: core/src/main/java/com/kumuluz/ee/EeApplication.java (hunk 2 above)
Comment (jmezna) [inline core/src/main/java/com/kumuluz/ee/EeApplication.java]: fixed

File: core/src/main/java/com/kumuluz/ee/EeApplication.java (hunk 3 above)
Comment (jmezna) [inline core/src/main/java/com/kumuluz/ee/EeApplication.java]: fixed

**Pull request merged**